bundles/org.openhab.binding.energenie/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,174 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="energenie"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="pm2lan">
+		<label>Gembird energenie EG-PM2-LAN</label>
+		<description>Gembird energenie programmable power strip with LAN interface</description>
+
+		<channels>
+			<channel id="socket1" typeId="socket1" />
+			<channel id="socket2" typeId="socket2" />
+			<channel id="socket3" typeId="socket3" />
+			<channel id="socket4" typeId="socket4" />
+		</channels>
+
+		<config-description>
+			<parameter name="host" type="text" required="true">
+				<label>IP-Address</label>
+				<description>The IP address of this device</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<label>Password</label>
+				<description>The password of this device</description>
+				<default>1</default>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<thing-type id="pmslan">
+		<label>Gembird energenie EG-PMS-LAN</label>
+		<description>Gembird energenie programmable power strip with LAN interface</description>
+
+		<channels>
+			<channel id="socket1" typeId="socket1" />
+			<channel id="socket2" typeId="socket2" />
+			<channel id="socket3" typeId="socket3" />
+			<channel id="socket4" typeId="socket4" />
+		</channels>
+
+		<config-description>
+			<parameter name="host" type="text" required="true">
+				<label>IP-Address</label>
+				<description>The IP address of this device</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<label>Password</label>
+				<description>The password of this device</description>
+				<default>1</default>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<thing-type id="pms2lan">
+		<label>Gembird energenie EG-PMS2-LAN</label>
+		<description>Gembird energenie programmable power strip with LAN interface</description>
+
+		<channels>
+			<channel id="socket1" typeId="socket1" />
+			<channel id="socket2" typeId="socket2" />
+			<channel id="socket3" typeId="socket3" />
+			<channel id="socket4" typeId="socket4" />
+		</channels>
+
+		<config-description>

```

Comment:
Put all these same configuration definitions in it's own file in `config` and refer to it, so you don't have to repeat it every time.

Reply:
Done with channel-groups

Reply:
You can use `config-description-ref` instead of `config-description` directly. In the thing you refer to the definition that then has to be specified in one place only. In a xml file in the `config` folder (next to `thing` folder.